app/src/main/java/com/example/criminalintent/presentation/crime_detail/CrimeFragment.kt

```diff
@@ -167,11 +204,20 @@ class CrimeFragment : Fragment(), FragmentResultListener {
         }
     }
 
-    private fun onActivityResult(result: ActivityResult) {
-        when (result.resultCode) {
-            RESULT_OK -> {
+    private fun onActivityResult(result: ActivityResult, requestCode: Int) {
+        when {
+            result.resultCode != RESULT_OK -> return
+            requestCode == REQUEST_CONTACT -> {
                 onContactSelected(result)
             }
+            requestCode == REQUEST_PHOTO -> {
+                updatePhotoView()
+                requireActivity().revokeUriPermission(
```

Comment:
The book mentions to call this in `onDetach()` too, to cover the possibility of an invalid response